trie/trie.go

```diff
@@ -355,7 +468,8 @@ func (t *Trie) insert(n node, prefix, key []byte, value node) (bool, node, error
 		if err != nil {
 			return false, nil, err
 		}
-		dirty, nn, err := t.insert(rn, prefix, key, value)
+		rn.setEpoch(epoch)
```

Comment:
may setEpoch(t.currentEpoch)

Reply:
this is not update, this is set the next node's epoch index